app/src/main/java/edu/calvin/cs262/teamc/activity_login.java

```diff
@@ -115,31 +120,42 @@ public void logIn(View view) {
 
         Intent logIn = new Intent(activity_login.this, MainActivity.class);
 
-        String requestUrl = String.format("https://calvincs262-fall2018-teamc.appspot.com/pinder/v1/login/%s/%s",usernameL.getText().toString(),passwordL.getText().toString());
-        StringRequest stringRequest = new StringRequest(Request.Method.GET, requestUrl, new Response.Listener<String>() {
+        String requestUrl =
+                String.format("https://calvincs262-fall2018-teamc.appspot.com/pinder/v1/login/%s/%s",
+                        usernameL.getText().toString(),passwordL.getText().toString());
+
+        StringRequest stringRequest = new StringRequest(
+                Request.Method.GET, requestUrl, new Response.Listener<String>() {
+
             @Override
             public void onResponse(String response) {
                 Intent logIn = new Intent(activity_login.this, MainActivity.class);
-                Log.e("onResponse",response);
                 JSONObject jsonObj = null;
                 try {
                     jsonObj = new JSONObject(response);
                     JSONArray arrJson = jsonObj.getJSONArray("items");
                     String responses = String.valueOf(arrJson);
-                    responses = responses.replace("[", "").replace("\"", "").replace("]", "");
-                    Log.e("loginJSON object", responses);
+                    responses = responses.replace("[", "")
+                            .replace("\"", "").
+                                    replace("]", "");
+
                     if (responses.equals("valid"))
                     {
                         logIn.putExtra("loginID",usernameL.getText().toString());
                         startActivity(logIn);
                         finish();
                     }
+                    else {
+                        Toast.makeText(activity_login.this,
+                                "Invalid username and password", Toast.LENGTH_SHORT).show();
+                    }
 
```

Comment:
Delete line (152) ?

Reply:
I kinda like it to make the if-else stand out more? Idk, what are your thoughts?

Reply:
That's fine. I think having just a bracket on a line (ex: line 151) has the same effect.